src/rpc/mempool.cpp

```diff
@@ -677,6 +677,8 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("incrementalrelayfee", ValueFromAmount(pool.m_incremental_relay_feerate.GetFeePerK()));
     ret.pushKV("unbroadcastcount", uint64_t{pool.GetUnbroadcastTxs().size()});
     ret.pushKV("fullrbf", pool.m_full_rbf);
+    ret.pushKV("permitbaremultisig", pool.m_permit_bare_multisig);
+    ret.pushKV("maxdatacarriersize", (pool.m_max_datacarrier_bytes ? *pool.m_max_datacarrier_bytes : 0));
```

Comment:
Is there a reason to name it something different here? Especially while the option is still broken and doesn't actually limit datacarriers?

Reply:
I was expecting this comment from you. :)

Named it same way as existing startup / config option (`-datacarriersize`), but I'm open to other names, as in fact it only limits `OP_RETURN`. What name seems to be good from Knots perspective? It would be good to not unnecessary break compatibility between Core and Knots.

Reply:
Well, you added "max" in front. I guess it makes sense, but might be surprising.

It's not clear what compatibility would mean before Core has fixed it to actually work.

Reply:
I slightly prefer it being called datacarriersize to match the config option, but don't feel strongly